tests/qat/test_quantum_backend.py

```diff
@@ -714,6 +714,163 @@ def test_create_duration_timeline_mapping(self):
             *[PostProcessing] * 4,
         ]
 
+    @pytest.mark.parametrize(
+        "hw", [get_test_model(), get_default_echo_hardware(), get_default_RTCS_hardware()]
+    )
+    def test_duration_timeline_times(self, hw):
+        # Tests that the creation of a duration timeline with a two-qubit circuit
+        # gives a timeline where each pulse is concurrent.
+        engine = hw.create_engine()
+        q1 = hw.get_qubit(0)
+        q2 = hw.get_qubit(1)
+
+        # Build a simple circuit
+        builder = (
+            get_builder(hw)
+            .X(q1, np.pi / 2.0)
+            .X(q2, np.pi)
+            .cnot(q1, q2)
+            .measure_mean_z(q1)
+            .measure_mean_z(q2)
+        )
+        res1 = engine.create_duration_timeline(builder.instructions)
+
+        # Check the times match up
+        for _, val in res1.items():
+            end = 0
+            for pos in val:
+                assert end == pos.start
+                end = pos.end
+
+    @pytest.mark.parametrize(
+        "hw", [get_test_model(), get_default_echo_hardware(), get_default_RTCS_hardware()]
+    )
+    def test_duration_timeline_sync(self, hw):
+        # Tests that a redundant sync has no effect on the circuit.
+        engine = hw.create_engine()
+        q1 = hw.get_qubit(0)
+        q2 = hw.get_qubit(1)
+
+        # Build a simple circuit
+        builder = (
+            get_builder(hw)
+            .X(q1, np.pi / 2.0)
+            .X(q2, np.pi)
+            .cnot(q1, q2)
+            .measure_mean_z(q1)
+            .measure_mean_z(q2)
+        )
+        res1 = engine.create_duration_timeline(builder.instructions)
+
+        # Build the same circuit with an unnecessary sync & check the times match up
+        builder = (
+            get_builder(hw)
+            .X(q1, np.pi / 2.0)
+            .X(q2, np.pi)
+            .synchronize([q1, q2])
+            .cnot(q1, q2)
+            .measure_mean_z(q1)
+            .measure_mean_z(q2)
+        )
+        res2 = engine.create_duration_timeline(builder.instructions)
+
+        for key in res1.keys():
+            starts1 = [
+                inst.start
+                for inst in res1[key]
+                if (not isinstance(inst.instruction, Delay) and inst.start != inst.end)
+            ]
+            starts2 = [
+                inst.start
+                for inst in res2[key]
+                if (not isinstance(inst.instruction, Delay) and inst.start != inst.end)
+            ]
+            ends1 = [
+                inst.start
+                for inst in res1[key]
+                if (not isinstance(inst.instruction, Delay) and inst.start != inst.end)
+            ]
+            ends2 = [
+                inst.start
+                for inst in res2[key]
+                if (not isinstance(inst.instruction, Delay) and inst.start != inst.end)
+            ]
+            assert starts1 == starts2
+            assert ends1 == ends2
```

Comment:
how the timeline contrast is tested here is fine but could improve:

look for a specific instruction other than a delay in both targets and make sure they start at the same time.
(measure/acquire would be a good example)
(and of course keep the delay case)

Reply:
To clarify, the test currently looks for all instructions that are _not_ Delays, and checks that they start and end at the same time as the instruction set without the sync. We ignore delays because the addition sync adds additional delay instruction to measure / acquire targets (which in practice can be squashed to a single delay for each target, but the result it the same).

However, we can remove `inst-start != inst.end` (I can't remember my reasoning for adding this in first place...).
Also added the check on acquires as you suggest, thanks!